# Subject: Classical Data Analysis

## Session 1 - Regression

### Exercise 2 


Implementation of Python StatsModels package with Quandl integration to calculate the Linear regression with one variable.  

Considering the OLS presented in Demo 3 (Outbound tourism statistics for Spain), develop a new regression analysis based on the Quandl dataset “Inbound tourism statistics for Spain”. This dataset is available at https://www.quandl.com/data/UTOR/INTUR_ESP

Use the field “Tourist arrivals at national borders – Thousands” as independent variable and the field “Travel - US$ Mn” as dependent variable.

- Interpret and discuss the OLS Regression Results. 
- Commit scripts in your GitHub account. You should export your solution code (.ipynb notebook) and push it to your repository “ClassicalDataAnalysis”.

The following are the tasks that should complete and synchronize with your repository “ClassicalDataAnalysis” until October 13. Please notice that none of these tasks is graded, however it’s important that you correctly understand and complete them in order to be sure that you won’t have problems with further assignments.


## Import Quandl data using your API Key

In [1]:
import quandl
quandl.ApiConfig.api_key = 'pVXKJ5_LwAwVsmip19m_'

In [6]:
data1 = quandl.get('UTOR/INTUR_ESP')
data1.head()
import pandas as pd
data12=pd.read_csv("https://www.quandl.com/api/v3/datasets/UTOR/INTUR_ESP.csv?api_key=wagAy5tFsmUZ84CH3Ng8")
data12.head()

Date  Tourist arrivals at national borders - Thousands  \
0  2014-12-31                                           64995.0   
1  2013-12-31                                           60675.0   
2  2012-12-31                                           57464.0   
3  2011-12-31                                           56177.0   
4  2010-12-31                                           52677.0   

   Tourism expenditure in the country - US$ Mn  Travel - US$ Mn  \
0                                          NaN          65100.0   
1                                          NaN          62584.0   
2                                          NaN          57877.0   
3                                          NaN          62447.0   
4                                          NaN          54305.0   

   Passenger transport - US$ Mn  
0                           NaN  
1                           NaN  
2                           NaN  
3                           NaN  
4                           NaN

## Regression model with Statsmodels and without a constant

In [11]:
data12.columns

Index(['Date', 'Tourist arrivals at national borders - Thousands',
       'Tourism expenditure in the country - US$ Mn', 'Travel - US$ Mn',
       'Passenger transport - US$ Mn'],
      dtype='object')

In [15]:
import statsmodels.api as sm
X = data12["Tourist arrivals at national borders - Thousands"]
y = data12["Travel - US$ Mn"]
model = sm.OLS(y, X).fit()


In [16]:
predictions = model.predict(X)
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:        Travel - US$ Mn   R-squared:                       0.967
Model:                            OLS   Adj. R-squared:                  0.965
Method:                 Least Squares   F-statistic:                     551.8
Date:                Fri, 13 Oct 2017   Prob (F-statistic):           1.68e-15
Time:                        21:30:43   Log-Likelihood:                -209.65
No. Observations:                  20   AIC:                             421.3
Df Residuals:                      19   BIC:                             422.3
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
====================================================================================================================
                                                       coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------------------------------------
Tourist arrivals at national borders - Thousands     0.9010      0.038     23.491      0.000       0.821       0.981
==============================================================================
Omnibus:                        6.516   Durbin-Watson:                   0.171
Prob(Omnibus):                  0.038   Jarque-Bera (JB):                1.723
Skew:                           0.056   Prob(JB):                        0.422
Kurtosis:                       1.566   Cond. No.                         1.00
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

### Interpreting the Table 

Considering the high r-quared vzalue of 0.967, I would say that our model is pretty good. The linear constant is equal to 0.9010.

## Regression model with Statsmodels and with a constant

In [17]:
import statsmodels.api as sm
X = data12["Tourist arrivals at national borders - Thousands"]
y = data12["Travel - US$ Mn"]
X = sm.add_constant(X) 
model = sm.OLS(y, X).fit()
predictions = model.predict(X)
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:        Travel - US$ Mn   R-squared:                       0.814
Model:                            OLS   Adj. R-squared:                  0.803
Method:                 Least Squares   F-statistic:                     78.61
Date:                Fri, 13 Oct 2017   Prob (F-statistic):           5.50e-08
Time:                        21:33:38   Log-Likelihood:                -203.23
No. Observations:                  20   AIC:                             410.5
Df Residuals:                      18   BIC:                             412.5
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
====================================================================================================================
                                                       coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------------------------------------
const                                            -3.832e+04   9518.711     -4.026      0.001   -5.83e+04   -1.83e+04
Tourist arrivals at national borders - Thousands     1.6339      0.184      8.866      0.000       1.247       2.021
==============================================================================
Omnibus:                        1.016   Durbin-Watson:                   0.368
Prob(Omnibus):                  0.602   Jarque-Bera (JB):                0.788
Skew:                          -0.130   Prob(JB):                        0.674
Kurtosis:                       2.063   Cond. No.                     3.33e+05
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 3.33e+05. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

### Interpreting the Table 

It is still quite a good model, 0.814 r-Squared. The model constants are -3.832e to the 4th and the linaer constant is 1.6339.